tensorflow/core/kernels/mkl_fused_batch_norm_op.cc

```diff
@@ -776,64 +935,67 @@ class MklFusedBatchNormOp : public OpKernel {
     const size_t kSavedVarianceIndex = 4;
     const size_t kReservedSpaceIndex = 5;
 
-    // allocate batch mean output tensor
+    // Allocate batch mean output tensor.
     MklDnnShape mkl_shape_batch_mean;
     mkl_shape_batch_mean.SetMklTensor(false);
     AllocateOutputSetMklShape(context, kBatchMeanIndex, batch_mean_tensor,
                               tf_shape_scale, mkl_shape_batch_mean);
-    CHECK_NOTNULL(*batch_mean_tensor);
-    // set NAN mean value in case of empty input tensor
+    DCHECK(*batch_mean_tensor);
+
+    // Set NAN mean value in case of empty input tensor
     int num_elements = tf_shape_scale.num_elements();
     auto batch_mean_data = (*batch_mean_tensor)->flat<U>().data();
     std::fill_n(batch_mean_data, num_elements, static_cast<U>(NAN));
 
-    // allocate batch variance output tensor
+    // Allocate batch variance output tensor.
     MklDnnShape mkl_shape_batch_variance;
     mkl_shape_batch_variance.SetMklTensor(false);
     AllocateOutputSetMklShape(context, kBatchVarianceIndex,
                               batch_variance_tensor, tf_shape_scale,
                               mkl_shape_batch_variance);
-    CHECK_NOTNULL(*batch_variance_tensor);
-    // set NAN variance value in case of empty input tensor
+    DCHECK(*batch_variance_tensor);
+
+    // Set NAN variance value in case of empty input tensor
     auto batch_variance_data = (*batch_variance_tensor)->flat<U>().data();
     std::fill_n(batch_variance_data, num_elements, static_cast<U>(NAN));
-
     // Mean and variance (without Bessel's correction) saved for backward
     // computation to serve as pre-computed mean and variance.
     MklDnnShape mkl_shape_saved_mean;
     mkl_shape_saved_mean.SetMklTensor(false);
     AllocateOutputSetMklShape(context, kSavedMeanIndex, saved_mean_tensor,
                               tf_shape_scale, mkl_shape_saved_mean);
-    CHECK_NOTNULL(*saved_mean_tensor);
-    // set 0 mean value in case of empty input tensor
+    DCHECK(*saved_mean_tensor);
+
+    // Set NAN mean value in case of empty input tensor
     auto saved_mean_data = (*saved_mean_tensor)->flat<U>().data();
-    std::fill_n(saved_mean_data, num_elements, static_cast<U>(0));
+    std::fill_n(saved_mean_data, num_elements, static_cast<U>(NAN));
 
     MklDnnShape mkl_shape_saved_variance;
     mkl_shape_saved_variance.SetMklTensor(false);
     AllocateOutputSetMklShape(context, kSavedVarianceIndex,
                               saved_variance_tensor, tf_shape_scale,
                               mkl_shape_saved_variance);
-    CHECK_NOTNULL(*saved_variance_tensor);
-    // set 0 variance value in case of empty input tensor
+    DCHECK(*saved_variance_tensor);
+
+    // Set NAN variance value in case of empty input tensor
     auto saved_variance_data = (*saved_variance_tensor)->flat<U>().data();
-    std::fill_n(saved_variance_data, num_elements, static_cast<U>(0));
+    std::fill_n(saved_variance_data, num_elements, static_cast<U>(NAN));
```

Comment:
Keep the 0 per PR https://github.com/tensorflow/tensorflow/pull/33444?
```suggestion
    std::fill_n(saved_variance_data, num_elements, static_cast<U>(0));
```